core/src/main/java/org/infinispan/marshall/exts/OptionalExternalizer.java

```diff
@@ -0,0 +1,38 @@
+package org.infinispan.marshall.exts;
+
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.util.Util;
+import org.infinispan.marshall.core.Ids;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Optional;
+import java.util.Set;
+
+public class OptionalExternalizer extends AbstractExternalizer<Optional> {
```

Comment:
Our ReadWriteEntryView exposes `Optional<V>`, so a user could decide to return that from a call to `eval(K key, Function<ReadWriteEntryView>...`, so I think we need to support that. Indeed, we should provide versions for various primitives. I'll add those.